File: Stack/dailyTemperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int b[] = new int[temperatures.length];
        Stack st = new Stack<>();
        for (int i = temperatures.length - 1; i >= 0; i--) {
            while (!st.isEmpty() && temperatures[st.peek()] <= temperatures[i]) {
                st.pop();
            }
            if (!st.isEmpty())
                b[i] = st.peek() - i;
            else
                b[i] = 0;
            st.push(i);
        }
        return b;
    }
}

// https://leetcode.com/problems/daily-temperatures/

File: Stack/dailyTemperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        s = []
        res = [0] * len(temperatures)

        for i in range(len(temperatures) - 1, -1, -1):
            while (len(s) > 0 and temperatures[s[-1]] <= temperatures[i]):
                s.pop()
            if (len(s) > 0):
                res[i] = s[-1] - i
            else:
                res[i] = 0
            s.append(i)
        return res
        